### ISSUE TITLE
How to specify notmuch `withEmacs=true` in systemPackages, and expose this to Emacs?

### ISSUE THREAD

**Issue by stig** [LnL7/nix-darwin#342]
TL;DR: I'd like to install Notmuch and specify the `withEmacs=true` option, and use that notmuch.el  in Emacs. 

The reason is that currently the version of notmuch.el I've got installed from melpa and my notmuch cli program is out of sync, and some commands don't work. Indeed [notmuch-emacs](https://notmuchmail.org/notmuch-emacs/) specifically mentions this:

> It is not recommended to install Notmuch Emacs from the Emacs Lisp Package Archive (ELPA), as the version there is likely not in sync with the command line interface.

My first attempt (that didn't yield a syntax error...) was:

```
  environment.systemPackages = with pkgs; [
    # other packages elided
    emacs
    (notmuch.override { withEmacs = true;})
  ];
```

But that didn't seem to do what I expect: notmuch.el is not installed into notmuch's share/emacs/site-lisp dir.

I then tried making a global override instead:

```
  nixpkgs.config.packageOverrides = pkgs: {
    notmuch = pkgs.notmuch.override { withEmacs = true; };
  };
```

This didn't work either.

What _did_ kinda work was this: 
> `nix-env -iA nixpkgs.notmuch --arg withEmacs true`

This installed `/nix/store/z739yicixjfc6a1m22534zni5lz05w9m-notmuch-0.32.1-emacs` (note the `-emacs` suffix) and that has notmuch.el, among others, in its share/emacs/site-lisp subdir.

(Aside: I have no idea how to delete that derivation, as `nix-env -e notmuch --arg withEmacs true` just deletes notmuch, not the one with the `-emacs` suffix 🤷🏼)

Regardless, I'd like to install this via systemPackages, and **I would love some help with this**. 

(My next challenge would be to tell Emacs about notmuch's share/emacs/site-lisp directory, so notmuch.el and the notmuch cli tool is always in sync. Hints about that too very welcome!)

**Comment by stig** [comment]
(I'm not sure Nix Darwin is the right place to ask, to be honest, but it's what I'm using.)

**Comment by stig** [comment]
Turns out this is what I needed to do: https://github.com/stig/dotfiles/commit/7b383e4315effe37f9fe3622eb73b915aac790aa

**Comment by Kaligule** [comment]
@stig  Thanks a lot. I had the same problem.

**Comment by lyderichti59** [comment]
Alternatively to @stig's solution, I managed to make it work without using an intermediary environment variable.

I made it work by adding both `pkgs.notmuch` & `pkgs.notmuch.emacs` to my list of nix packages.
This makes both CLIs available (`notmuch` & `notmuch-emacs-mua`  respectively).

The embedded emacs code lies in `notmuch-emacs-mua`'s nix store folder, under `"/nix/store/<sha256>-notmuch-<version>-emacs/share/emacs/site-lisp"`. By using `which`  and `realpath`, you can retrieve this path dynamically from the current binary (tolerant to updates) and use it as the load-path of the emacs lib.

See below how I load notmuch in emacs :
```elisp
(use-package notmuch
  :load-path (lambda () (shell-command-to-string "realpath -m \"$(which notmuch-emacs-mua)/../../share/emacs/site-lisp/\" | tr -d '\n'"))
  ...)
```
